### ISSUE TITLE
optimizer definition on line 55 in dgl/examples/pytorch/gnn_explainer/explain_main.py is wrong

### ISSUE THREAD

**Issue by sharique1006** [dmlc/dgl#3944]
## 🐛 Bug

In `dgl/examples/pytorch/gnn_explainer/explain_main.py` on line 55, you have defined the optimizer as 
`optim = th.optim.Adam(explainer.parameters(), lr=args.lr, weight_decay=args.wd)`

According to the paper, model entropy H(Y) in the objective function is constant as the model is fixed. We only want to optimize the masks.
Since you are passing explainer.parameters(), the optimizer will also optimize the model parameters as the model parameters are present in explainer.parameters(). This will change the model as well as can lead to exploding prediction logits when training the explainer.

Instead, you should define it as 
`optim = th.optim.Adam([explainer.edge_mask, explainer.node_feat_mask], lr=args.lr, weight_decay=args.wd)`

This is how it has also been defined in the source code of GNNExplainer.
You can check the same here on line 614, 622 
[https://github.com/RexYing/gnn-model-explainer/blob/master/explainer/explain.py](url)

**Comment by mufeili** [comment]
Nice catch! Could you open a PR to fix the issue?

CC @zhjwy9343 for awareness.
